src/components/Wowerlay.tsx

```diff
@@ -8,85 +7,106 @@ import {
   onMounted,
   ref,
   shallowRef,
+  toRef,
   watch,
 } from 'vue';
-import { cWowerlayAnimEnter, cWowerlayAnimLeave, cWowerlayBackground } from '../consts';
+import { useFloating, flip, shift, offset, autoUpdate, Middleware, Side } from '@floating-ui/vue';
 
-import type { WowerlayTemplateRef } from './WowerlayRenderer';
-import type { WowerlayBaseProps } from './WowerlayReusables';
-import { WowerlayRenderer } from './WowerlayRenderer';
-import { isElement } from '../utils';
-import { wowerlayBaseProps } from './WowerlayReusables';
-
-export type { WowerlayTemplateRef } from './WowerlayRenderer';
-
-const ATTR_PREFIX = 'data-wowerlay-';
-const SCOPE_ATTR_QUERY = `[${ATTR_PREFIX}scope]`;
-const STOP_ATTR_QUERY = `[${ATTR_PREFIX}stop]`;
-
-export interface WowerlayProps extends WowerlayBaseProps {
-  visible: boolean;
-}
+import { Props } from './Wowerlay.constants';
+import { NOOP, isElement } from '../utils';
+import { attrs, syncSize } from './Wowerlay.middlewares';
 
 export const Wowerlay = defineComponent({
   name: 'Wowerlay',
   inheritAttrs: false,
-  props: {
-    ...wowerlayBaseProps,
-    visible: {
-      required: true,
-      type: Boolean as PropType<WowerlayProps['visible']>,
-    },
-  },
+  props: Props,
   emits: {
     'update:visible': null! as (value: boolean) => void,
     'update:el': null! as (value: HTMLElement | null) => void,
   },
-  setup(props, { emit, expose }) {
-    const wowerlayInstance = shallowRef<WowerlayTemplateRef>();
+  setup(props, { emit }) {
+    const popoverEl = shallowRef<HTMLDivElement | null>(null);
+    const backgroundEl = shallowRef<HTMLElement | null>();
+
+    const { floatingStyles } = useFloating(
+      toRef(props, 'target'), //
```

Comment:
Empty comment

Reply:
this was for preventing prettier to make it single line, but it's now longer so you're right.